crates/cli/src/js.rs

```diff
@@ -48,4 +66,232 @@ impl JS {
         )?;
         Ok(compressed_source_code)
     }
+
+    pub fn exports(&self) -> Result<Vec<String>> {
+        let module = self.parse_module()?;
+
+        // function foo() ...
+        let mut functions = HashMap::new();
+        // export { foo, bar as baz }
+        let mut named_exports = vec![];
+        // export function foo() ...
+        let mut exported_functions = vec![];
+        for item in module.body {
+            match item {
+                ModuleItem::ModuleDecl(ModuleDecl::ExportDecl(ExportDecl {
+                    decl: Decl::Fn(f),
+                    ..
+                })) => {
+                    if !f.function.params.is_empty() {
+                        bail!("Exported functions with parameters are not supported");
+                    }
+                    if f.function.is_generator {
+                        bail!("Exported generators are not supported");
+                    }
+                    exported_functions.push(f.ident.sym);
+                }
+                ModuleItem::ModuleDecl(ModuleDecl::ExportNamed(e)) => {
+                    for specifier in e.specifiers {
+                        if let ExportSpecifier::Named(n) = specifier {
+                            let orig = match n.orig {
+                                ModuleExportName::Ident(i) => i.sym,
+                                ModuleExportName::Str(s) => s.value,
+                            };
+                            let exported_name = n.exported.map(|e| match e {
+                                ModuleExportName::Ident(i) => i.sym,
+                                ModuleExportName::Str(s) => s.value,
+                            });
+                            named_exports.push((orig, exported_name));
+                        }
+                    }
+                }
+                ModuleItem::Stmt(Stmt::Decl(Decl::Fn(f))) => {
+                    functions.insert(
+                        f.ident.sym,
+                        (f.function.params.is_empty(), f.function.is_generator),
+                    );
+                }
+                _ => continue,
+            }
+        }
+
+        let mut named_exported_functions = named_exports
+            .into_iter()
+            .filter_map(|(orig, exported)| {
```

Comment:
This is a little complicated because we have 3 separate cases for named exports. They can be:
- not a function
- a function we don't support (has arguments or is a generator)
- a function we support

And we need to drop the first case, error on the second, and figure out the exported name for the third case.